### PR TITLE
feat: Added ref method to TensorFlow Frontend

### LINKED ISSUE #26140: ref

### LINKED ISSUE #26516: ivy.ref
**Why should this be implemented?**
- it's needed for a complex/long frontend function implementation

**Links to native framework implementations**
- [TensorFlow](https://www.tensorflow.org/api_docs/python/tf/Tensor#ref) (`ref` instance method of `tf.Tensor`)

### PR DESCRIPTION
**Pull request by ShruAgarwal**
# PR Description
Implemented an [Instance Method](https://www.tensorflow.org/api_docs/python/tf/Tensor#ref) named, `ref` to TensorFlow Frontend
> Quick info about `ref`:
Returns a hashable reference object to the Tensor.
*Example:* 
>>x = tf.constant(5)
>>x.ref()
True

**I tested running the function over all backends but, unfortunately, it fails for all of them except for `tensorflow`. Later, I came to know that `ref` method or any such similar method doesn't exist in any other backend frameworks.** 😅

## Related Issue
Pending issue to be resolved #26516

Close #26140

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

Please review the changes and let me know your thoughts if I'm correct or not. 🙏

### CODE REVIEW

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hi, seems like you have changed `docs/demos` which you might have done by mistake. Can you revert that change because we can't merge with that change in place?

**Comment by Kamalesh3112** [comment]
About to start

File: ivy/functional/frontends/tensorflow/tensor.py [hunk 1]
```diff
@@ -53,6 +53,9 @@ def shape(self):
     def get_shape(self):
         return tf_frontend.raw_ops.Shape(input=self)
 
+    def ref(self):
+        return self._ivy_array.ref()
+
     def set_shape(self, shape):
         if shape is None:
             return
```
Comment (Kamalesh3112) [inline ivy/functional/frontends/tensorflow/tensor.py:61]: def ref(self, shape):
      if shape is None:
           return self._ivy_array.ref()

**Review by Kamalesh3112 — COMMENTED** [review]
I have added the ref() function method for the code number lines  59, 60 and 61.  It requires ref() method to be added into the code for Tensorflow Frontend application for the issue #26316

File: ivy/functional/frontends/tensorflow/tensor.py (hunk 1 above)
Comment (ShruAgarwal) [reply ivy/functional/frontends/tensorflow/tensor.py:61]: Hey @Kamalesh3112 
Thanks for reviewing! :)

But, according to the TensorFlow API docs, `ref` method doesn't have any parameters. 😅

Did you add that by mistake or does it point to something that needs to be improved?

**Comment by ShruAgarwal** [comment]
> Hi, seems like you have changed `docs/demos` which you might have done by mistake. Can you revert that change because we can't merge with that change in place?

@KareemMAX Yeah, I didn't touch that file 😕

I tried resolving but it shows that there is nothing to change or commit 👇

git status

**Check the `git status` message in the screenshot above.**
Please let me know what I should do now. 😅

**Comment by ShruAgarwal** [comment]
@KareemMAX Sorry for those extra revert commits as I'm learning to figure it out. 😅

Also, do the failed checks from above need to be resolved?

**Comment by ShruAgarwal** [comment]
@KareemMAX or @Infrared1029

**I request to kindly please review the changes I've made and let me know your thoughts!** 😅🙂

File: ivy/functional/frontends/tensorflow/tensor.py (hunk 1 above)
Comment (Kamalesh3112) [reply ivy/functional/frontends/tensorflow/tensor.py:61]: Yeah, the ref() method doesn't have any parameters and it has to be improved by adding those particular lines of code.

**Comment by NiteshK84** [comment]
Hi @ShruAgarwal , 
please resolve the merge conflicts so we can proceed with your pull request.

**Comment by NiteshK84** [comment]
Hi @ShruAgarwal , 
please resolve the merge conflicts so we can proceed with your pull request.

**Comment by ShruAgarwal** [comment]
> Hi @ShruAgarwal , please resolve the merge conflicts so we can proceed with your pull request.

Hey @NiteshK84,
This pull request was made to solve issue #26140, but it has been closed as it was not a priority to be solved (months ago). Do you think I should still make changes? 😅

**Comment by KareemMAX** [comment]
Hello,

I think this PR is unactive for a little while, so I'll close it for now. If you want to continue working on it in the future please feel free to open it and start working.

Thanks and happy contributing 😊

**Pull request closed without merging**